src/components/QuizBuilder.ts

```diff
@@ -0,0 +1,67 @@
+import { multipleChoiceQuestions } from "../questions/multipleChoiceQuestions";
+import { singleChoiceQuestions } from "../questions/singleChoiceQuestions";
+import { textInputQuestions } from "../questions/textInputQuestions";
+
+export interface Question {
+    question: string;
+    answers: string[];
+    correctAnswers: string[];
+};
+
+class Quiz {
+    questions: Question[] = [];
+    score: number = 0;
+};
+
+export class QuizBuilder {
+    buildQuiz(): Quiz {
+        const quiz: Quiz = new Quiz();
+        let questions = [];
+
+        questions = questions.concat(this.poseQuestion(multipleChoiceQuestions));
+        questions = questions.concat(this.poseQuestion(singleChoiceQuestions));
+        questions = questions.concat(this.poseQuestion(textInputQuestions));
+        
+        quiz.questions = this.shuffle(questions);
+        return quiz;
+    };
+
+    /* pick questions for the quiz */
```

Comment:
Although I like you describing your methods I think you should have a quick look into https://jsdoc.app/about-getting-started.html .. you will find a convention for documention